### PR TITLE
Make extensions rtp_module configurable

### PR DESCRIPTION
**Pull request by Rados13**

### CODE REVIEW

File: lib/membrane_webrtc_plugin/extension/vad.ex [hunk 1]
```diff
@@ -24,7 +28,10 @@ defmodule Membrane.WebRTC.Extension.VAD do
   def get_uri(), do: @uri
 
   @impl true
-  def get_rtp_module(vad_id), do: %@rtp_module{vad_id: vad_id}
+  def get_rtp_module(vad_id, rtp_opts) do
+    rtp_opts = Map.new(rtp_opts)
+    Map.merge(%@rtp_module{vad_id: vad_id}, rtp_opts)
```
Comment (balins) [inline lib/membrane_webrtc_plugin/extension/vad.ex:33]:
```suggestion
    struct!(@rtp_module, [vad_id: vad_id | rtp_opts])
```

File: lib/membrane_webrtc_plugin/extension.ex [hunk 2]
```diff
@@ -7,9 +7,20 @@ defmodule Membrane.WebRTC.Extension do
   alias ExSDP.Attribute.{FMTP, Extmap}
   alias Membrane.{RTP, WebRTC}
 
-  @type t :: module()
+  defstruct module: __MODULE__,
+            rtp_opts: Keyword.new()
```
Comment (balins) [inline lib/membrane_webrtc_plugin/extension.ex:11]:
```suggestion
  @enforce_keys [:module]
  defstruct @enforce_keys ++ [rtp_opts: Keyword.new()]
```

File: lib/membrane_webrtc_plugin/extension.ex [hunk 3]
```diff
@@ -7,9 +7,20 @@ defmodule Membrane.WebRTC.Extension do
...
+
+  @type t :: %__MODULE__{
+          module: module(),
+          rtp_opts: Keyword.t()
+        }
   @type maybe_t :: t() | :not_supported
 
+  @doc """
+  Create a `__MODULE__` struct for specific extension.
```
Comment (balins) [inline lib/membrane_webrtc_plugin/extension.ex:20]:
```suggestion
  Creates a `#{inspect(__MODULE__)}` struct for a specific extension.
```

**Review by balins — CHANGES_REQUESTED** [review]

**Comment by balins** [comment]
It don't think it needs to be included in this PR, but at some point we should revisit if it would be beneficial to replace some callbacks with additional fields in `%WebRTC.Extension{}` struct (e.g. `uri`) and try to handle as much as possible in base `WebRTC.Extension` module. This approach could help us mitigate repeating code among all modules implementing `WebRTC.Extension`.

File: lib/membrane_webrtc_plugin/extension.ex [hunk 4]
```diff
@@ -7,9 +7,20 @@ defmodule Membrane.WebRTC.Extension do
...
   alias Membrane.{RTP, WebRTC}
 
-  @type t :: module()
+  defstruct module: __MODULE__,
+            rtp_opts: Keyword.new()
+
+  @type t :: %__MODULE__{
+          module: module(),
+          rtp_opts: Keyword.t()
+        }
```
Comment (mickel8) [inline lib/membrane_webrtc_plugin/extension.ex:16]: We need docs here. Also, we need better type for `rtp_opts`

**Review by mickel8 — CHANGES_REQUESTED** [review]

File: lib/membrane_webrtc_plugin/extension.ex (hunk 4 above)
Comment (Rados13) [reply lib/membrane_webrtc_plugin/extension.ex:16]: I am not sure if we can specify here a better description for rtp_opts, as every RTP module has its own options.

File: lib/membrane_webrtc_plugin/extension.ex [hunk 5]
```diff
@@ -2,22 +2,31 @@ defmodule Membrane.WebRTC.Extension do
   @moduledoc """
   A module that provides mappings between `ExSDP.Attribute.Extmap` and modules implementing
   `Membrane.WebRTC.Extension` behaviour.
+
+  This struct contains:
+    * module - extension specific module which implement `Membrane.WebRTC.Extension` behaviour
+    * rtp_opts - options for RTP module. This allows configuring RTP module to your needs
+    * uri - a URI that identifies extension in SDP
+    * name - an atom identifying the extension in `Membrane.RTP.SessionBin`
```
Comment (mickel8) [inline lib/membrane_webrtc_plugin/extension.ex:10]: Put this as doc for type `t` i.e.
```elixir
@typedoc """
...
''''
@type t :: %__MODULE__{
```

**Review by mickel8 — APPROVED** [review]

**Review by balins — APPROVED** [review]
Just a suggestion, we can also consider mapping extension modules like 
```extensions:  Enum.map(opts.extensions, &(if is_struct(&1), do: &1, else: &1.new()))```
in `WebRTC.EndpointBin.handle_init/1` to avoid passing lists like `[TWCC.new(), Mid.new(), Rid.new()]` when we don't want to customize extensions.

**Pull request merged**